File: appcompat-sample/src/main/java/com/kfaraj/support/appcompat/sample/MainFragment.java
package com.kfaraj.support.appcompat.sample;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

/**
 * Demonstrates how to use the AppCompat library.
 */
public class MainFragment extends Fragment {

    /**
     * Constructor.
     */
    public MainFragment() {
        super(R.layout.fragment_main);
    }

    /**
     * Creates a new instance of this fragment class.
     *
     * @return a new instance of this fragment class.
     */
    @NonNull
    public static MainFragment newInstance() {
        return new MainFragment();
    }

}

File: appcompat-sample/src/main/java/com/kfaraj/support/appcompat/sample/MainFragment.kt
package com.kfaraj.support.appcompat.sample

import androidx.fragment.app.Fragment

/**
 * Demonstrates how to use the AppCompat library.
 */
class MainFragment : Fragment(R.layout.fragment_main) {

    companion object {

        /**
         * Creates a new instance of this fragment class.
         */
        fun newInstance(): MainFragment {
            return MainFragment()
        }

    }

}
